### DISCUSSION TITLE
How OpenFAST calculate the pitching moment at the blade root?

### DISCUSSION THREAD

**Discussion by sstonee2024** [OpenFAST/openfast#2521] [Q&A] [answered]
Dear all.

I would like to understand how the "RootMzb1" output, which represents the "Blade 1 pitching moment at the blade root," is calculated in OpenFAST. Could you please share if there is a specific equation or methodology used for this calculation?

Thank you so much.
J.C.

**Comment by jjonkman** [comment]
Dear @sstonee2024,

See a similar question asked and answered in the following OpenFAST issue: https://github.com/OpenFAST/openfast/issues/1916.

Best regards,

**Reply by sstonee2024** [reply]
Thank you for your prompt reply, Mr. Jonkman.

I reviewed the post you mentioned and found that the equation discussed there relates to the blade-root bending moments, not the pitching moment that I am seeking.

Additionally, I examined the source file ElastoDyn.f90 and discovered only two references to the pitching moment. I attempted to understand the code within ElastoDyn.f90, but it contains more than 10,000 lines, which is overwhelming for me.

I recognize that I should dig deeper into that file, but I find it quite challenging. Could you please do me a favor by providing the equation for the pitching moment in OpenFAST? I don't know if my request is too much, I apologize if I bothered you.

Thank you very much.
Best regards,

**Reply by jjonkman** [reply, marked as answer]
Dear @sstonee2024,

The equation summarized in the issue I linked to above applies to both the blade-root bending moments and the pitching moment.  That is, once the integral along the blade of the applied aerodynamic pitching moments + ( radial vector ) x ( applied aerodynamic forces + gravitation forces - inertia forces ) is calculated, the resulting moment vector is projected along the x, y, and z axes of the blade-root coordinate system to calculate the bending moments about x and y and the pitching moment about z.

For the mathematical details, see the ElastoDyn Theory Manual on readthedocs: https://openfast.readthedocs.io/en/main/source/user/elastodyn/index.html.  The _FASTLoads.doc_ contains the mathematical details for the blade-root loads, although you'd need the review the documents earlier in the list to understand the nomenclature.

Best regards,

**Reply by sstonee2024** [reply]
Dear @jjonkman,

Thank you so much for your quick reply, Mr. Jonkman. It helped me a lot. Truly appreciate it.

Wish you everything going well.

Best regards,
